packages/rich-text-editor/src/RichTextEditor/controlmap.ts

```diff
@@ -65,132 +65,106 @@ function liftOrIndentList(action: "lift" | "indent"): Command {
 export function buildControlMap(
   schema: Schema,
   editorState: EditorState,
-  controls: ToolbarControls[][]
+  controls?: ToolbarItems[]
 ) {
+  if (!controls) return {}
+  const toolbarControls = {}
   // creates an initial array of objects for the controls to be sorted into
-  const toolbarControls = Array.from({ length: controls.length }, () => ({}))
-  // console.log("toolbar initial", toolbarControls)
-
-  const controlGroupings = controls.reduce(
-    (prevSection, section, sectionIndex) => {
-      const innerControls = section.map(item => [item, sectionIndex])
-      if (innerControls && innerControls.length) {
-        return [...prevSection, ...innerControls]
-      }
-      return [...prevSection]
-    },
-    []
-  )
-
-  // console.log("controlGroupings", controlGroupings)
-  const controlsSections = new Map(controlGroupings)
-  // console.log("control section Map", controlsSections)
-
-  // const ideaObj = [
-  //   {
-  //     bold: {
-  //       isActive: markIsActive(editorState, type),
-  //       action: toggleMarkCommand(type),
-  //       label: "Bold",
-  //       icon: boldIcon,
-  //     },
-  //     italic: {
-  //       isActive: markIsActive(editorState, type),
-  //       action: toggleMarkCommand(type),
-  //       label: "Italic",
-  //       icon: italicIcon,
-  //     },
-  //   },
-  //   {
-  //     underline: {
-  //       isActive: markIsActive(editorState, type),
-  //       action: toggleMarkCommand(type),
-  //       label: "Underline",
-  //       icon: underlineIcon,
-  //     },
-  //   },
-  //   {
-  //     bullet_list: {
-  //       action: toggleListCommand(schema.nodes.bullet_list),
-  //       isActive: false,
-  //       label: "Bullet List",
-  //       icon: bulletListIcon,
-  //     },
-  //   },
-  // ]
+  const controlGroups = controls.reduce((prev, current) => {
+    if (!current?.name) return prev
+    return { ...prev, [current.name]: current.group }
+  }, {})
+
+  console.log("controlGroups", controlGroups)
+  console.log("toolbarControls", toolbarControls)
 
   if (schema.marks.bold) {
     const type = schema.marks.bold
-    const groupIndex = controlsSections.get("bold")
-    toolbarControls[groupIndex]["bold"] = {
-      isActive: markIsActive(editorState, type),
-      action: toggleMarkCommand(type),
-      label: "Bold",
-      icon: boldIcon,
+    toolbarControls[controlGroups["bold"]] = {
```

Comment:
Might be worth putting `controlGroups["bold"]` into a variable to make it a little easier to read